docs/upgrade-to-v4.md

```diff
@@ -0,0 +1,110 @@
+# Upgrade to V4
+
+Sequelize V4 is a major release and it introduces newer features, breaking changes and deprecations. Majority of sequelize codebase has been refactored to use ES2015 features. The following guide lists some of the changes to upgrade from v3 to v4. See [Changelog](http://docs.sequelizejs.com/manual/changelog/changelog.html#future) for full list of changes.
```

Comment:
1. Changelog link can be simply replaced with `[Changelog](manual/changelog)`

2. Starting line should say `Sequelize V4 is the next major release and it introduces new features, breaking changes and deprecations`